test/CMakeLists.txt

```diff
@@ -28,6 +22,17 @@ if(LIBCORO_FEATURE_NETWORKING)
     )
 endif()
 
+if(LIBCORO_FEATURE_THREADING)
+    list(APPEND LIBCORO_TEST_SOURCE_FILES
+        bench.cpp
+        test_semaphore.cpp
+        test_io_scheduler.cpp
+        test_ring_buffer.cpp
```

Comment:
I believe the usage of `coro::io_scheduler` in this file can also be replaced with `coro::thread_pool` 1 thread

Reply:
I wasn't able to get this one working -- the program hung:

```
22:21:31: Starting test\libcoro_test.exe...
Randomness seeded to: 2625223567
Acquiring lock
lock acquired, emplacing back 1
coroutine done
block task acquiring lock
block task acquired lock, waiting on event
id = 1 waiting to acquire the lock
id = 2 waiting to acquire the lock
id = 3 waiting to acquire the lock
id = 4 waiting to acquire the lock
set task setting event
id = 4 lock acquired
id = 4 coroutine done
id = 3 lock acquired
id = 3 coroutine done
id = 2 lock acquired
id = 2 coroutine done
id = 1 lock acquired
id = 1 coroutine done
produce: 1
produce: 2
produce: 3
consume: 1
produce: 4
consume: 2
produce: 5
consume: 3
produce: 6
consume: 4
produce: 7
consume: 5
produce: 8
consume: 6
produce: 9
consume: 7
produce: 10
consume: 8
consume: 9
consume: 10
```

Reply:
Interesting, I'll mark it as something to look at deeper, we can keep it as is for now.

Reply:
Oops, I think I was just impatient. It does pass, but takes a while to run. Does this sound OK to you?

Reply:
It does do 10 million produce and consumes, so should take a little time but not a crazy amount.. if you run it a few more times is it consistently long? It's possible the yield is choking it out and the 50ms yield previously was making it complete a lot faster...

Reply:
Here's an example from CI (the Ubuntu 22.04 run):

```
Thu, 12 Oct 2023 03:32:59 GMT 2: consume: 10
Thu, 12 Oct 2023 03:34:04 GMT 2: ===============================================================================
```

It takes about 1 minute to run this last line on Ubuntu.

Reply:
I'd maybe run it before and after the switch and time it, that'll tell us if we made a major regression in the speed.

Reply:
Before the switch on my PC, it takes a few seconds at most. After, a while (guessing 1 minute?).

Reply:
I can put this in a separate PR if you'd like to look at it. For now, it's in a separate branch: https://github.com/bjones1/libcoro/tree/slow-threading-refactor.

Reply:
Here's the CI from Fedora-37; again, about a minute:

```
Thu, 12 Oct 2023 03:32:36 GMT 2: consume: 10
Thu, 12 Oct 2023 03:33:24 GMT 2: ===============================================================================
```

Reply:
Looking at [earlier CI where this test was still included](https://github.com/jbaldwin/libcoro/actions/runs/6461866284/job/17542416231), it looks like that's a pretty standard time:

```
Mon, 09 Oct 2023 21:30:13 GMT 2: consume: 10
Mon, 09 Oct 2023 21:31:05 GMT 2: Acquiring semaphore
```

Reply:
So this should be fine to include. I'll go ahead and add it to the PR.

Reply:
Ok great, thanks for checking